src/pages/images/actions/forms/ImportImageFromTarballForm.tsx

```diff
@@ -0,0 +1,146 @@
+import { ChangeEvent, FC, useState } from "react";
+import { useEventQueue } from "context/eventQueue";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import {
+  changeImagePublicity,
+  createImageAlias,
+  createImageFromTarball,
+} from "api/images";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { Link, useNavigate } from "react-router-dom";
+
+interface Props {
+  close: () => void;
+  project: string;
+}
+
+const ImportImageFromTarballForm: FC<Props> = ({ close, project }) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const [fileList, setFileList] = useState<FileList | null>(null);
+
+  const notifySuccess = () => {
+    const published = (
+      <Link to={`/ui/project/${project}/images`}>published</Link>
+    );
+    toastNotify.success(<>Image {published} from file.</>);
+  };
+
+  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      const files = e.target.files;
+      setFileList(files);
```

Comment:
instead of managing file with additional state, you can use `formik.setFieldValue` here

Reply:
I actually did try this before this implementation and it did not work. I also read somewhere online that for files, it is unsupported. I recreated it again just in case I might have done it wrong, but I got the same error: 
![image](https://github.com/user-attachments/assets/8287ae60-880f-4e46-bc51-85e1d74898d9)

For reference, this was the new implementation:
```
  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
    if (e.target.files) {
      formik.setFieldValue("fileList", e.target.files);
    }
  };
```

Reply:
You can refer to #853 for how this can be done. The key thing here is to remove `formik.getFieldProps` from the file input and let `changeFile` handle selection changes